airflow/utils/log/file_task_handler.py

```diff
@@ -517,9 +524,13 @@ def _read_from_logs_server(self, ti, worker_log_rel_path) -> tuple[list[str], li
             if response.text:
                 messages.append(f"Found logs served from host {url}")
                 logs.append(response.text)
+        except NoHostnameForWorkerLogs:
+            logger.debug("Could not fetch logs from worker because no hostname on TI object %s", ti)
```

Comment:
Is it worth providing some actionable advice here? When I have seen this issue before it was due to an incorrect `hostname_callable` configuration I believe.